### PR TITLE
weechat: update to 1.6 & offer --without-tcl

### PR DESCRIPTION
**Pull request by azrdev**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

**Comment by mistydemeo** [comment]
Looks like the build is failing; are you able to look into it?

**Comment by azrdev** [comment]
@mistydemeo thx for the headsup – I'm a linuxbrew user only, so I dunno why this fails here, but I reported upstream at https://github.com/weechat/weechat/issues/276#issuecomment-252416756

File: Formula/weechat.rb [hunk 1]
```diff
@@ -39,6 +40,7 @@ def install
     args << "-DENABLE_LUA=OFF" if build.without? "lua"
     args << "-DENABLE_PERL=OFF" if build.without? "perl"
     args << "-DENABLE_RUBY=OFF" if build.without? "ruby"
+    args << "-DENABLE_TCL=OFF" if build.without? "tcl"
```
Comment (MikeMcQuaid) [inline Formula/weechat.rb]: Is this not defaulting to off, then, and changing behavour?

Comment (MikeMcQuaid) [inline Formula/weechat.rb]: Let's leave `tcl` on by default, thanks.

**Comment by MikeMcQuaid** [comment]
@azrdev Any news on the comments? Thanks!

**Comment by azrdev** [comment]
@MikeMcQuaid If I understand [`build.with(out)`](http://www.rubydoc.info/github/Homebrew/brew/master/BuildOptions#with%3F-instance_method) correctly, It might suffice to use `option "without-tcl"` on line 18 instead of `option "with-tcl"` -- correct?
Apart from that, [I see here](http://www.rubydoc.info/github/Homebrew/brew/master/Formula#depends_on-class_method) only the possibility to add tcl as a `:recommended` (i.e. default ON) dependency

**Comment by MikeMcQuaid** [comment]
> It might suffice to use option "without-tcl" on line 18 instead of option "with-tcl" -- correct?

Yup 👍

**Comment by dunn** [comment]
@BrewTestBot test this please

**Comment by dunn** [comment]
https://bot.brew.sh/job/Homebrew%20Core/11685/
```
Undefined symbols for architecture x86_64:
  "_res_9_init", referenced from:
      _irc_server_xfer_send_ready_cb in irc-server.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by dunn** [comment]
Also if you could change the test from
```ruby
system "weechat", "-r", "/quit"
```
to
```ruby
system bin/"weechat", "-r", "/quit"
```
that'd be great!

**Comment by azrdev** [comment]
@dunn done, whatever this has to do with this PR

**Comment by robbiet480** [comment]
@azrdev I was able to get this to build finally with the changes proposed in #4427.

**Comment by neutric** [comment]
Upstream added resolv (see [here](https://github.com/weechat/weechat/commit/e098a0dc5414f67006c9e8f0ff02f22f715a0a55) and [here](https://github.com/weechat/weechat/commit/228040dc04648507d9a57dcc6324d436cd97a4bc)) to the build scripts but compilation continues to fail with undefined symbols (see dunn's comment above on Nov 18). `export LDFLAGS=-lresolv` as suggested in #4427 solves the problem. Upstream knows that this issue persists but doesn't have macOS to debug and thus [asks for help](https://github.com/weechat/weechat/issues/276#issuecomment-267806781). 

CMake properly detects resolv, so something is probably wrong about where in the CMake script it is added, but I know too little about CMake to be of help.

**Comment by MikeMcQuaid** [comment]
@neutric Convincing upstream to set `LDFLAGS=-lresolv` in CMake for macOS is probably the best fix.

**Pull request closed without merging**